modules/hostap/src/supp_main.c

```diff
@@ -489,13 +486,13 @@ static void interface_handler(struct net_mgmt_event_callback *cb,
 
 	if (mgmt_event == NET_EVENT_IF_ADMIN_UP) {
 		LOG_INF("Network interface %d (%p) up", net_if_get_by_iface(iface), iface);
-		submit_iface_work(ctx, iface, add_interface);
+		add_interface(get_default_context(), iface);
```

Comment:
The reason why it was done like this is that the net_mgmt thread stack can be quite small by default and this can cause issues if we do bigger operations like this from it. So I am not convinced this is a good idea. What kind of issues are you seeing when using a work queue?

Reply:
For instance, once the net_mgmt callback is completed, the interface can be removed completely, but the reference is used in this worker thread. So, we need a lock to synchronize network stack vs. worker thread.